### ISSUE TITLE
Detected N+1 Query  

### ISSUE THREAD

**Issue by emresasi** [jorenvanhee/joren.co#6]
Hello, the code really helped me and works well. Thank you for the code. However, there is a problem, it re-runs the query as many records as there are in the related table and logs 

[2023-09-15 18:30:14] local.INFO: Detected N+1 Query
[2023-09-15 18:30:14] local.INFO: Model: App\Models\Order
Relation: App\Models\OrderDetail
Num-Called: 2
Call-Stack:

It prints this code. How can we prevent this problem?

The way I run the code:

    public function update(OrderRequest $request, Order $order)
    {

        $validated = $request->validated();
        $order->update($validated['order']);

        $order->orderDetails()->createUpdateOrDelete($validated['orderDetails']);

    }

**Comment by jorenvanhee** [comment]
You could probably change the behaviour in `updateOrCreateRecords` method from looping over the records and using `updateOrCreate`, to something that first fetches all the records with a `whereIn` query and then inserts the records that were not found.

But to be honest, in most cases this n+1 will not be an issue. It's not like an index page with 20 blog posts with each 100 comments resulting in 2000 queries. In our case, if you send 20 rows to the `createUpdateOrDelete` it will execute 20 queries + some additional create queries. It does not exponentially grow.

tldr: leave it as is, if it ever becomes a problem (which it will probably not) you can always change the underlaying implementation :)

**Comment by emresasi** [comment]
Thank you, I will leave it like this for now as you said.
